airflow/contrib/hooks/aws_glue_job_hook.py

```diff
@@ -0,0 +1,128 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+from airflow.exceptions import AirflowException
+from airflow.contrib.hooks.aws_hook import AwsHook
+import time
+from botocore.exceptions import ClientError
+
+
+class AwsGlueJobHook(AwsHook):
+    """
+    Interact with AWS Glue - create job
+
+    :param job_name: unique job name per AWS account
+    :type str
+    :param desc: job description
+    :type str
+    :param aws_conn_id: aws connection id
+    :type aws_conn_id: str
+    """
+
+    def __init__(self,
+                 job_name=None,
+                 desc=None,
+                 aws_conn_id='aws_default',
+                 *args, **kwargs):
+        self.job_name = job_name
+        self.desc = desc
+        self.aws_conn_id = aws_conn_id
+        super(AwsGlueJobHook, self).__init__(self.aws_conn_id, *args, **kwargs)
+
+    def get_conn(self):
+        return self.get_client_type('glue')
+
+    def list_jobs(self):
+        conn = self.get_conn()
+        return conn.get_jobs()
+
+    def initialize_job(self, script_arguments=None):
+        """
+        Initializes connection with AWS Glue
+        to run job
+        :return:
+        """
+        glue_client = self.get_conn()
+
+        try:
+            job_response = self.get_glue_job()
+            job_name = job_response['Name']
+            job_run = glue_client.start_job_run(
+                JobName=job_name,
+                Arguments=script_arguments
+            )
+            return self.job_completion(job_name, job_run['JobRunId'])
+        except ClientError as general_error:
+            raise AirflowException(
+                'Failed to run aws glue job, error: {error}'.format(
+                    error=str(general_error)
+                )
+            )
+
+    def job_completion(self, job_name=None, run_id=None):
+        """
+        :param job_name:
+        :param run_id:
+        :return:
+        """
+        glue_client = self.get_conn()
+
+        while True:
+
+            job_status = glue_client.get_job_run(
+                JobName=job_name,
+                RunId=run_id,
+                PredecessorsIncluded=True
+            )
+            job_run_state = job_status['JobRun']['JobRunState']
+
+            if job_run_state in {'FAILED', 'STOPPED', 'SUCCEEDED'}:
+                self.log.info("Exiting Job {} Run State: {}"
+                              .format(run_id, job_run_state))
+                return {'JobRunState': job_run_state, 'JobRunId': run_id}
+            else:
+                self.log.info("Polling for AWS Glue Job {} current run state"
+                              .format(job_name))
+                time.sleep(6)
+
+    def get_glue_job(self):
```

Comment:
this code can only create a job but I want to run exists job by name
so suggest to use
```
    def get_or_create_glue_job(self):
        try:
            glue_client = self.get_conn()
            glue_client.get_job(JobName=self.job_name)
        except ClientError as client_err:
            if client_err.response['Error']['Code'] == \
                    'EntityNotFoundException':
                return self._create_glue_job()
            raise client_err

    def _create_glue_job(self):
        glue_client = self.get_conn()

        if self.s3_bucket is None:
            raise AirflowException(
                'Could not initialize glue job, '
                'error: Specify Parameter `s3_bucket`'
            )
        self.log.info("Now creating and running AWS Glue Job")

        s3_log_path = "s3://{bucket_name}/{logs_path}{job_name}"\
            .format(bucket_name=self.s3_bucket,
                    logs_path=self.S3_GLUE_LOGS,
                    job_name=self.job_name)

        execution_role = self.get_iam_execution_role()
        script_location = self._check_script_location()
        create_job_response = glue_client.create_job(
            Name=self.job_name,
            Description=self.desc,
            LogUri=s3_log_path,
            Role=execution_role['Role']['RoleName'],
            ExecutionProperty={"MaxConcurrentRuns": self.concurrent_run_limit},
            Command={"Name": "glueetl", "ScriptLocation": script_location},
            MaxRetries=self.retry_limit,
            AllocatedCapacity=self.num_of_dpus
        )
        return create_job_response

```

and then in `initialize_job` do


```
        glue_client = self.get_conn()
        try:
            self.get_or_create_glue_job()
            job_run = glue_client.start_job_run(
                JobName=self.job_name,
                Arguments=script_arguments
            )
            self.log.info("Run job with id: {}".format(job_run['JobRunId']))
            return self.job_completion(self.job_name, job_run['JobRunId'])
        except Exception as general_error:
            raise AirflowException(
                'Failed to run aws glue job, error: {error}'.format(
                    error=str(general_error)
                )
            )